doc/computing/randomness.rst

```diff
@@ -0,0 +1,315 @@
+
+The `random_state` parameter determines whether multiple calls to :term:`fit`
+(for estimators) or to :term:`split` (for cv splitters) will produce the same
+results, according to these rules:
+
+- If an int is passed, calling `fit` or `split` multiple times yields the
+  same results.
+- If `None` or a `RandomState` instance is passed: `fit` and `split` will
+  yield different results each time they are called, producing an event chain
+  of maximal entropy. `None` is the default value for all `random_state`
+  parameters.
+
+  .. note::
+      Since passing `random_state=None` is equivalent to passing the global
+      `RandomState` instance from `numpy`
+      (`random_state=np.random.mtrand._rand`), we will not explicitly mention
+      `None` here, but everything that applies to instances also applies to
+      using `None`.
+
+.. warning:: TL;DR
+
+    Unless you know what you are doing, we recommend to use integers as the
+    `random_state` parameter of estimators and cv splitters, as it is usually
+    the safest option and it allows you to isolate different sources of
+    variance in your analysis. Leaving the default (`None`) or using
+    `RandomState` instances is sometimes useful to assess to statistical
+    significance of your results, but can also have surprising effects.
+
+Getting reproducible results across multiple executions
+-------------------------------------------------------
+
+In order to obtain reproducible (i.e. constant) results across multiple
+program executions, the easiest way is to set a `rng` variable to an
+integer value at the top of your program, and pass it down to any object that
+accepts a `random_state` parameter::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import train_test_split
+
+    >>> rng = 42  # or any other integer
+    >>> X, y = make_classification(random_state=rng)
+    >>> rf = RandomForestClassifier(random_state=rng)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ...                                                     random_state=rng)
+    >>> rf.fit(X_train, y_train).score(X_test, y_test)
+    0.88
+
+We are now guaranteed that the result of this script will always be 0.88, no
+matter how many times we run it. Changing the global `rng` variable to a
+different value should affect the results, as expected.
+
+We could have used a `RandomState` instance instead of an integer::
+
+    >>> import numpy as np
+    >>> rng = np.random.RandomState(0)
+
+This too would give us reproducible results across executions. However, using
+`RandomState` instances may have surprising and unwanted effects that will be
+described later.
+
+We do not recommend either setting the global `numpy` seed by calling
+`np.random.seed(0)`. See `here
+<https://stackoverflow.com/questions/5836335/consistently-create-same-random-numpy-array/5837352#comment6712034_5837352>`_
```

Comment:
you could also add the link of NEP 19 on numpy random generators https://numpy.org/neps/nep-0019-rng-policy.html#numpy-random. Although we might not want to confuse users with the new numpy random API as it is not yet used in scikit-learn.

Reply:
Thanks @albertcthomas . Yes I'm afraid linking to the SLEP might confuse users further since it's about Generators, and since they recommend passing RandomState instances (things are more subtle for us)